java_fundamentals_block/main_task/src/main/java/Actions.java

```diff
@@ -0,0 +1,112 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Random;
+
+public class Actions { //Main tasks in Java Fundamentals block
+
+    private static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public static void helloUser() throws IOException {
+        System.out.println("Enter your name:");
+        String userName = reader.readLine();
+        System.out.println("Hello, " + userName +"!");
+    }
+
+    public static void arrayReverse() throws IOException {
+        System.out.println("Enter the sequence separated by a space:");
+        String sequence = reader.readLine();
+        String[] elements = sequence.split(" ");
+
+        for (int i = elements.length - 1; i >= 0; i--) {
+            System.out.print(elements[i] +" ");
+        }
+        System.out.println();
+    }
+
+    public static void randomNumbers() throws IOException {
+        System.out.println("Enter the number of elements:");
+        int numberOfElements = Integer.parseInt(reader.readLine());
+        int[] array = new int[numberOfElements];
+        Random random = new Random();
+
+        for (int i = 0; i < numberOfElements; i++) {
+            array[i] = random.nextInt();
+        }
+        for (int element : array) {
+            System.out.println(element);
+        }
+        for (int element : array) {
+            System.out.print(element + " ");
+        }
+        System.out.println();
+    }
+
+    public static void sumAndMultiple() throws IOException {
+        int sum = 0;
+        int multiple = 1;
+
+        System.out.println("Enter the numbers:");
+        while (true) {
+            String input = reader.readLine();
+            if (!input.equals("")) {
+                int number = Integer.parseInt(input);
+                sum += number;
+                multiple *= number;
+            }
+            else {
+                break;
+            }
+        }
+        System.out.println("The sum of all numbers is " + sum);
+        System.out.println("The product of all numbers is " + multiple);
```

Comment:
product? is it typo?

Reply:
No. The product of numbers is the result you get when you multiply them together.